core/soong_kernel_module.go

```diff
@@ -78,16 +90,42 @@ var soongKbuildRule = apctx.StaticRule("bobKbuild",
 	}, "kmod_build", "depfile", "extra_includes", "extra_cflags", "kbuild_extra_symbols", "kernel_dir", "kernel_cross_compile",
 	"kbuild_options", "make_args", "output_module_dir", "cc_flag", "hostcc_flag", "clang_triple_flag")
 
+
 func (m *kernelModuleBackend) GenerateAndroidBuildActions(ctx android.ModuleContext) {
 	builtModule := android.PathForModuleOut(ctx, m.Name()+".ko")
 	symvers := android.PathForModuleOut(ctx, "Module.symvers")
 
+	// preserve symvers location for this module (for the parent pass)
+	m.Symvers = symvers
```

Comment:
OK, so as we process the build actions for each module we save the path to Module.symvers, and pick up the path in the walk. So we're relying on the order of the GenerateAndroidBuildAction calls to be be in dependency order (which I believe is true).

This doesn't feel like the proper way to do this, but is OK for now.

Reply:
Ok.